### PR TITLE
Samples are starting at the boundaries of the global AABB

### PR DESCRIPTION
**Pull request by chevtche**

### CODE REVIEW

**Comment by chevtche** [comment]
retest this please

File: livre/eq/render/RayCastRenderer.cpp [hunk 1]
```diff
@@ -164,6 +164,14 @@ struct RayCastRenderer::Impl
         tParamNameGL = glGetUniformLocation( program, "invModelViewMatrix" );
         glUniformMatrix4fv( tParamNameGL, 1, false, frustum.getInvModelViewMatrix( ).array );
 
+        tParamNameGL = glGetUniformLocation( program, "globalAabbMin" );
```
Comment (bilgili) [inline livre/eq/render/RayCastRenderer.cpp]: globalAABBMin

Comment (bilgili) [inline livre/eq/render/RayCastRenderer.cpp]: Add a note while you are dividing by two.

File: livre/eq/render/shaders/fragRayCast.glsl [hunk 2]
```diff
@@ -124,21 +126,27 @@ void main( void )
...
+        AABB globalAabb = AABB( globalAabbMin, globalAabbMax );
 
         float tnear, tfar;
         intersectBox( eye, aabb, tnear, tfar );
 
+        float tnearGlobal, tfarGlobal;
+        intersectBox( eye, globalAabb, tnearGlobal, tfarGlobal );
+
         vec3 nearPlaneNormal = vec3( 0.0f, 0.0f, 1.0f );
         float tNearPlane = dot( nearPlaneNormal, vec3( 0.0, 0.0, -nearPlaneDist )) / dot( nearPlaneNormal, normalize( pixelEyeSpacePos.xyz ));
```
Comment (bilgili) [inline livre/eq/render/shaders/fragRayCast.glsl]: line too long error

**Comment by bilgili** [comment]
add a changelog to notify which fix number you have fixed. I did not do that before myself on the previous 3 fixes, but you can do it :)

**Comment by chevtche** [comment]
I am a bit lost... There is no issue opened for this one and also this issue was not existing in the previous release because it was introduced by this recent nyquist limit computation

Should I write in the changeog something like: "Samples start from the boundaries of the global AABB"

???

**Comment by chevtche** [comment]
Ready to merge ?? :)

File: livre/eq/render/RayCastRenderer.cpp [hunk 3]
```diff
@@ -164,6 +164,16 @@ struct RayCastRenderer::Impl
         tParamNameGL = glGetUniformLocation( program, "invModelViewMatrix" );
         glUniformMatrix4fv( tParamNameGL, 1, false, frustum.getInvModelViewMatrix( ).array );
 
+        // Because the volume is centered to the origin we can compute the volume AABB by using
+        // the volume total size.
+        Vector3f halfWorldSize = _volInfo.worldSize / 2.0;
```
Comment (bilgili) [inline livre/eq/render/RayCastRenderer.cpp]: const

File: doc/Changelog.md [hunk 4]
```diff
@@ -6,6 +6,8 @@ Changelog {#Changelog}
 
 # Release 0.4 (2015-11-09) {#Release040}
 
+* [#217](https://github.com/BlueBrain/Livre/pull/217):
+  Improve #177 by moving the first ray's samples on the boundary of the volume AABB
```
Comment (bilgili) [inline doc/Changelog.md]: line too long error

**Comment by chevtche** [comment]
Updated

**Comment by bilgili** [comment]
+1

**Pull request merged**